src/hotspot/cpu/riscv/c1_LIRGenerator_riscv.cpp

```diff
@@ -772,7 +772,7 @@ void LIRGenerator::do_ArrayCopy(Intrinsic* x) {
   ciArrayKlass* expected_type = nullptr;
   arraycopy_helper(x, &flags, &expected_type);
   if (x->check_flag(Instruction::OmitChecksFlag)) {
-    flags = 0;
+    flags = (flags & LIR_OpArrayCopy::unaligned);
```

Comment:
Should be LIR_OpArrayCopy::unaligned|LIR_OpArrayCopy::overlapping? See below.

Reply:
fixed.